### ISSUE TITLE
OpenBSD fails to boot on bhyve on AMD hardware

### ISSUE THREAD

**Issue by akole** [joyent/smartos-live#927]
When booting bhyve branded VM with OpenBSD 6.6 install iso the process fails with:

{"log":"vm exit[0]\n\treason\t\tSVM\n\trip\t\t0x00000000fffee1ec\n\tinst_length\t0\n\texitcode\t","stream":"stderr","time":"2020-04-23T17:27:36.533033000Z"}

SmartOS platform is joyent_20200408T231825Z
Server is a Proliant DL385 G6 with dual AMD Opterons

**Comment by sjorge** [comment]
Have you tried adding `vmadm update UUID bhyve_extra_opts=-w"` it's needed on intel too to not make OpenBSD trip bhyve by accessing unsupported MSRs

**Comment by akole** [comment]
> 
> 
> Have you tried adding `vmadm update UUID bhyve_extra_opts="-w"` it's needed on intel too to not make OpenBSD trip bhyve by accessing unsupported MSRs

Yes, I set these in the spec json and both seem to be there when checking vmadm info:
  "bootrom": "uefi",
  "bhyve_extra_opts": "-w"

**Comment by sjorge** [comment]
Ack, should work then... so it's probably a SVM vs VMX thing.

**Comment by akole** [comment]
> 
> 
> Ack, should work then... so it's probably a SVM vs VMX thing.

Anything I can do to help debug this issue?

FreeBSD host with NetBSD VM also seemed to have similar issues few years back if it's any help:
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=203643

**Comment by despair86** [comment]
oh hell i posted more info in https://github.com/joyent/illumos-joyent/issues/248

**Comment by despair86** [comment]
@akole this should be fixed in the latest platform image
